### PR TITLE
Search the specific guest attributes from host by ovirt-shell

### PR DESCRIPTION
**Pull request by hkx303**
**Descriptiono**
Added `guest_search()` , `get_rhevm_info()` and guest_ip related function

**Test Result**
```
>>>    cli = RHEVMCLI(server, ssh_user, ssh_pwd, admin_user, admin_pwd)
>>>    print(cli.guest_search(guest_name, host_ip, host_user, host_pwd))
{'guest_name': '8.3_Server_x86_64', 'guest_ip': '10.73.5.227', 'guest_uuid': 'e27cd702-0e74-4ddf-8755-c2a04207372a', 'guest_state': 'up', 'vdsm_uuid': 'd7dc6478-d0e5-4c22-a61c-7a76d5507de4', 'vdsm_hwuuid': '5CEB2C71-0C8F-434C-93A1-9E1D61AF9B09', 'vdsm_hostname': 'bootp-73-5-203.rhts.eng.pek2.redhat.com', 'vdsm_version': 'vdsm-4.30.51-1.el7ev', 'vdsm_cpu': '2', 'vdsm_cluster': 'Default'}
```

### CODE REVIEW

File: hypervisor/virt/rhevm/rhevmcli.py [hunk 1]
```diff
@@ -88,3 +88,98 @@ def info(self):
...
+        :return: the mac address for the guest
+        """
+        cmd = f"ovirt-shell -c -E 'list nics --parent-vm-name {guest_name} --show-all' | grep  '^mac-address'"
+        ret, output = self.ssh.runcmd(cmd)
+        if not ret and "mac-address" in output:
+            mac_addr = output.strip().split(': ')[1].strip()
+            logger.info(f"rhevm({self.server}) guest mac is: {mac_addr}")
+            return mac_addr
+        else:
+            logger.info(f"Failed to check rhevm({self.server}) guest mac")
```
Comment (Junefen) [inline hypervisor/virt/rhevm/rhevmcli.py:183]: Here if need to return a null value? I guess if the "cmd = f"nmap -sP -n {gateway} | grep -i -B 2 {guest_mac} | {option}" in def get_guest_ip_by_mac will fail with a None {guest_mac}...

File: hypervisor/virt/rhevm/rhevmcli.py [hunk 2]
```diff
@@ -88,3 +88,98 @@ def info(self):
...
+        cmd = f"ovirt-shell -c -E 'show {object_type} {object_id}' |grep '^{value}'"
+        ret, output = self.ssh.runcmd(cmd)
+        if not ret and value in output:
+            result = output.strip().split(':')[1].strip()
+            logger.info(f"Succeeded to get rhevm {object_type} ({object_id}) {value}: {result}")
+            return result
+        else:
+            logger.info(f"Failed to get rhevm {object_type} ({object_id}) {value}")
+
+    def get_guest_ip_by_mac(self, guest_name, host_ip, host_user, host_pwd):
```
Comment (Junefen) [inline hypervisor/virt/rhevm/rhevmcli.py]: Maybe the function name should be ```get_guest_ip``` or ```get_guest_ip_by_name```

File: hypervisor/virt/rhevm/rhevmcli.py [hunk 3]
```diff
@@ -88,3 +88,98 @@ def info(self):
...
+        :return: the value you want to get
+        """
+        cmd = f"ovirt-shell -c -E 'show {object_type} {object_id}' |grep '^{value}'"
+        ret, output = self.ssh.runcmd(cmd)
+        if not ret and value in output:
+            result = output.strip().split(':')[1].strip()
+            logger.info(f"Succeeded to get rhevm {object_type} ({object_id}) {value}: {result}")
+            return result
+        else:
+            logger.info(f"Failed to get rhevm {object_type} ({object_id}) {value}")
```
Comment (Junefen) [inline hypervisor/virt/rhevm/rhevmcli.py:129]: Is it better to return a null value when failed to get the info? Or it returns None, right?

Comment (hkx303) [reply hypervisor/virt/rhevm/rhevmcli.py:129]: Changed :)

File: hypervisor/virt/rhevm/rhevmcli.py (hunk 1 above)
Comment (hkx303) [reply hypervisor/virt/rhevm/rhevmcli.py:183]: return None if it fails.

File: hypervisor/virt/rhevm/rhevmcli.py (hunk 2 above)
Comment (hkx303) [reply hypervisor/virt/rhevm/rhevmcli.py]: Changed to `get_guest_ip`

**Comment by Junefen** [comment]
Looks good :)

**Pull request merged**